includes/polling/os/windows.inc.php

```diff
@@ -8,179 +8,116 @@
-
-    if (strstr($poll_device['sysDescr'], 'Build 3790')) {
-        $version = 'Server 2003 Datacenter (NT 5.2)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 6001')) {
-        $version = 'Server 2008 Datacenter (NT 6.0)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 6002')) {
-        $version = 'Server 2008 Datacenter SP2 (NT 6.0)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 7600')) {
-        $version = 'Server 2008 Datacenter R2 (NT 6.1)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 7601')) {
-        $version = 'Server 2008 Datacenter R2 SP1 (NT 6.1)';
-    }
-
-    if (strstr($poll_device['sysDescr'], 'Build 9200')) {
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 9200')) {
         $version = 'Server 2012 Datacenter (NT 6.2)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 7601')) {
+        $version = 'Server 2008 Datacenter R2 SP1 (NT 6.1)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 7600')) {
+        $version = 'Server 2008 Datacenter R2 (NT 6.1)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 6002')) {
+        $version = 'Server 2008 Datacenter SP2 (NT 6.0)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 6001')) {
+        $version = 'Server 2008 Datacenter (NT 6.0)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 3790')) {
+        $version = 'Server 2003 Datacenter (NT 5.2)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build 2195')) {
+        $version = '2000 Datacenter Server (NT 5.0)';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build Number: 1381')) {
+        $version = 'NT Datacenter 4.0';
+    } elseif (str_contains($poll_device['sysDescr'], 'Build Number: 1057')) {
+        $version = 'NT Datacenter 3.51';
     }
+}//end version if
 
-    if (strstr($poll_device['sysDescr'], 'Build 9600')) {
-        $version = 'Server 2012 R2 (NT 6.3)';
-    }
-}//end if
-
-if (strstr($poll_device['sysDescr'], 'Uniprocessor')) {
-    $features = 'Uniprocessor';
-}
-
-if (strstr($poll_device['sysDescr'], 'Multiprocessor')) {
+if (str_contains($poll_device['sysDescr'], 'Multiprocessor')) {
     $features = 'Multiprocessor';
+} elseif (str_contains($poll_device['sysDescr'], 'Uniprocessor')) {
+    $features = 'Uniprocessor';
 }
 
-// Detect processor type? : I.E.  x86 Family 15 Model 2 Stepping 7
 // Detect Dell hardware via OpenManage SNMP
 $hw = snmp_get($device, '.1.3.6.1.4.1.674.10892.1.300.10.1.9.1', '-Oqv', 'MIB-Dell-10892');
-$hw = trim(str_replace('"', '', $hw));
-if ($hw) {
-    $hardware = 'Dell '.$hw;
-}
+$hw = trim($hw, ' "');
+if (!empty($hw)) {
+    $hardware = 'Dell ' . $hw;
 
-$serial = snmp_get($device, '.1.3.6.1.4.1.674.10892.1.300.10.1.11.1', '-Oqv', 'MIB-Dell-10892');
-$serial = trim(str_replace('"', '', $serial));
+    $serial = snmp_get($device, '.1.3.6.1.4.1.674.10892.1.300.10.1.11.1', '-Oqv', 'MIB-Dell-10892');
+    $serial = trim($serial, ' "');
```

Comment:
I can't figure out why the str_replace is done i tested on snmp 5.5 and 5.7 and there are no quotes in the output of a normal snmpget....